utils/hoursUtils.js

```diff
@@ -0,0 +1,131 @@
+const airtableClient = require('../airtableClient')
+
+
+module.exports = {
```

Comment:
I'm sure there could be improvements here, open to suggestions

Reply:
Generally, `moment.js` used to be the go-to for date and time manipulation and checks, and you have a _lot_ of that here.  It wouldn't be a bad idea to use that to reduce the amount of utility functions you are re-implementing (i.e. isOpenNow checks).

It also has i18n support out of the box, which may be useful for supporting the language settings.

Also, there's some formats here that are probably coming back from AIrtable that would need to be documented in terms of what is expected from that API (much like the bus routes).

Reply:
I heard that moment.js is considered legacy and no longer recommended: https://momentjs.com/docs/#/-project-status/
Temporal will be coming in JS but is still in proposal: https://github.com/tc39/proposal-temporal...not sure how fast those proposals go but figured we'd move to that....if that's gonna take forever, could try something like Luxon though? (One of the recommendations from moment). Right now we don't support multiple timezones either, a library of some sort could probably help with that down the line too. 

That's a good point about language settings though... however... would that require passing some language config from the front end??

Hmm, also good point about formatting... like for `open_time_digits` kind of thing? Any idea where the best place for that would be?

Reply:
Luxon looks legit, so I'd give that a spin.  Temporal looks like a better expansion on the `Date` object, but I don't know how it handles the i18n bits most of the other libraries do.

For language settings, you can actually pull it from the browser, but [it seems dubious at best](https://phrase.com/blog/posts/detecting-browser-language-preference-with-javascript/) now that I look into it.  

For data structure formatting, you can do it in one of two places:

* In the Airtable service that pulls the hours down in the first place
* At the method level where you're processing the values (or just link to them)

There's probably a way with TypeScript being the norm where those models could be exposed publicly, and could see it in docs (Like `javadoc` essentially).  No idea about that though.

Reply:
Implemented Luxon. Yeah, I'm skeptical about pulling from browser. Which specific fields do you think should have format called out? the ones that I guess are most particular are the `open_time_digits` and `close_time_digits` which must be a 4-number string.

Yeah, in kotlin or java we would define pojos or data classes and would validate any formatting on the fields there. Not sure if there's a way or convention like that for javascript...

Reply:
Since most of JS is presentational, you don't worry about it as much.  For values being posted, however, if you're rolling TypeScript you probably do it with the similar patterns as Java.  I wouldn't know, as I have yet to work in a JS environment where that mattered.